http/resource.go

```diff
@@ -144,6 +148,27 @@ var resourcePatchHandler = withUser(func(w http.ResponseWriter, r *http.Request,
 		return http.StatusForbidden, nil
 	}
 
+	override := r.URL.Query().Get("override") == "true"
+	rename := r.URL.Query().Get("rename") == "true"
+
+	if !override && !rename {
+		if _, err = d.user.Fs.Stat(dst); err == nil {
+			return http.StatusConflict, nil
+		}
+	}
+
+	if rename {
+		suffix := "_copy"
+		_, err = d.user.Fs.Stat(dst)
+		exists := err == nil
+
+		for exists {
```

Comment:
I think better would be to rewrite it like:
```go
for {
	// ...
	if notExist {
		break;
	}
}
```
With that you can reduce code duplications